modules/core/02-client/types/exp/hooks.go

```diff
@@ -0,0 +1,56 @@
+package exp
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/v7/modules/core/exported"
+)
+
+// Hooks specifies methods to enable writing custom logic which executes
+// after specific client lifecycle methods.
+// Note: this interface is experimental and could be removed in favour of a different solution, requiring code changes.
+type Hooks interface {
+	// OnClientCreated is executed upon client creation.
+	OnClientCreated(ctx sdk.Context, clientID string) error
+	// OnClientUpdated is executed when a client is updated.
+	OnClientUpdated(ctx sdk.Context, clientID string, consensusHeights []exported.Height) error
+	// OnClientUpgraded is executed when a client is upgraded.
+	OnClientUpgraded(ctx sdk.Context, clientID string) error
+}
```

Comment:
I think a hook for Misbehaviour is also necessary. CCV has expressed interest in something like this

`OnClientMisbehaviour(ctx sdk.Context, clientID string, clientMsg exported.ClientMessage)`